api/v1/aerospikecluster_types.go

```diff
@@ -578,12 +589,19 @@ type AerospikeRoleSpec struct {
 
 // AerospikeUserSpec specifies an Aerospike database user, the secret name for the password and, associated roles.
 type AerospikeUserSpec struct {
+	// AerospikeAuthMode specifies an authentication mode (Internal or PKIOnly) enabled for the user.
+	// +kubebuilder:default:=Internal
+	// +optional
+	AerospikeAuthMode AerospikeAuthMode `json:"aerospikeAuthMode,omitempty"`
+
 	// Name is the user's username.
 	Name string `json:"name"`
 
 	// SecretName has secret info created by user. User needs to create this secret from password literal.
 	// eg: kubectl create secret generic dev-db-secret --from-literal=password='password'
-	SecretName string `json:"secretName"`
+	// required when AerospikeAuthMode is Internal
```

Comment:
```suggestion
     // SecretName has secret info created by user. User needs to create this secret from `password` literal.
	// It is a required field in case of Internal AerospikeAuthMode.
	// eg: kubectl create secret generic dev-db-secret --from-literal=password='password'
```

```